src/cdk-experimental/menu/menu.ts

```diff
@@ -67,80 +66,52 @@ import {MENU_TRIGGER, MenuTrigger} from './menu-trigger';
     {provide: CDK_MENU, useExisting: CdkMenu},
   ],
 })
-export class CdkMenu extends CdkMenuGroup implements Menu, AfterContentInit, OnInit, OnDestroy {
-  /**
-   * Sets the aria-orientation attribute and determines where menus will be opened.
-   * Does not affect styling/layout.
-   */
-  @Input('cdkMenuOrientation') orientation: 'horizontal' | 'vertical' = 'vertical';
-
+export class CdkMenu extends CdkMenuBase implements AfterContentInit, OnDestroy {
   /** Event emitted when the menu is closed. */
   @Output() readonly closed: EventEmitter<void | 'click' | 'tab' | 'escape'> = new EventEmitter();
 
-  /** Handles keyboard events for the menu. */
-  private _keyManager: FocusKeyManager<CdkMenuItem>;
-
-  /** Manages items under mouse focus. */
-  private _pointerTracker?: PointerFocusTracker<CdkMenuItem>;
-
   /** List of nested CdkMenuGroup elements */
+  // TODO(mmalerba): Does it make sense for menubars to have groups as well?
   @ContentChildren(CdkMenuGroup, {descendants: true})
   private readonly _nestedGroups: QueryList<CdkMenuGroup>;
 
-  /** All child MenuItem elements nested in this Menu. */
-  @ContentChildren(CdkMenuItem, {descendants: true})
-  private readonly _allItems: QueryList<CdkMenuItem>;
-
-  /** The Menu Item which triggered the open submenu. */
-  private _openItem?: CdkMenuItem;
-
   constructor(
     private readonly _ngZone: NgZone,
-    readonly _elementRef: ElementRef<HTMLElement>,
-    @Optional() @Inject(MENU_STACK) readonly _menuStack?: MenuStack,
+    elementRef: ElementRef<HTMLElement>,
+    @Optional() @Inject(MENU_STACK) menuStack?: MenuStack,
     @Optional() @Inject(MENU_TRIGGER) private _parentTrigger?: MenuTrigger,
     @Self() @Optional() @Inject(MENU_AIM) private readonly _menuAim?: MenuAim,
-    @Optional() private readonly _dir?: Directionality,
+    @Optional() dir?: Directionality,
   ) {
-    super();
-  }
-
-  ngOnInit() {
-    this._menuStack?.push(this);
+    super(elementRef, menuStack, dir);
+    // TODO(mmalerba): This is matching what we were already doing, but judging
```

Comment:
Do we want to report a reason on our `closed` Observable? It looks like the API signature was set up to support that but nothing actually gives a value other than undefined